manifests/init.pp

```diff
@@ -55,6 +56,12 @@
     $dhcp_interfaces = $interfaces
   }
 
+  if $dnsupdatekey {
+    $_dnsupdatekey_split    = split($dnsupdatekey, '[/]')
+    $_dnsupdatekey_basename = $_dnsupdatekey_split[-1]
+    $dnskeyname_real        = pick($dnskeyname, $_dnsupdatekey_basename)
+  }
```

Comment:
This needs an else case to set $dnskeyname_real to $dnskeyname if $dnsupdatekey is unset.